torch/csrc/lazy/tutorial.md

```diff
@@ -0,0 +1,251 @@
+
+
+```python
+
+print(torch.jit.last_executed_optimized_graph())
+
+# graph(%t : Tensor,
+#       %maybe : Tensor):
+#   %2 : Tensor = prim::profile[profiled_type=Float(1, strides=[1], requires_grad=0, device=cpu), seen_none=0](%t)
+#    = prim::profile()
+#   return (%2)
+```
+
+We could see that the if statement disappeared and jit trace only traced the `else` path. In fact, jit trace can trace **only** aten operations. It's completely oblivious to any control flow operations such as `if`, `for` or an exception.
+If this sounds unsafe to you, that's because it is!
+
+Let's now learn how we can solve this issue with Lazy Tensors.
+
+The first step is to move the inputs to the Lazy device. The Lazy device isn't any real hardware device. Your code still runs either on CPU or on GPU if you set `LTC_TS_CUDA="1"`.
+
+The lazy device is however very special: it makes PyTorch "remember" every aten operation (into a graph) the user calls rather than eagerly executing it. It's lazy that way ;) get it?
+
+So, the lazy device is API that users should use to trace their models with Lazy Tensor. It's also a PyTorch device which is a very convenient way for implementing tracing based on PyTorch dispatcher.
+
+First of all, we need a little bit of setup. The Lazy Tensor needs a backend to actually run traced graphs. We implemented a TorchScript-based backend to give our users end-to-end experience running their models with Lazy Tensor. It also serves as an example for hardware vendors looking to integrate with Lazy Tensor. 
+
+
+```
+import torch._lazy
+import torch._lazy.ts_backend
+torch._lazy.ts_backend.init()
+```
+
+Now, we can our example,
+
+```
+dev = "lazy"
+t_lazy = torch.ones(1).to(dev)
+maybe_false_lazy = torch.BoolTensor([0]).to(dev)
+lazy_result = add_two_maybe(t_lazy, maybe_false_lazy)
+```
+
+This is pretty cool! Eventually, however, we would still like to execute our computation and access the result, wouldn't we? 
+
+There are a number of ways to do it. Typically, PyTorch transparently triggers the execution when the user tries to access the result e.g. print a tensor out, move the tensor to a non-lazy device, etc.
+
+Let's give it a try:
+
+```
+lazy_result = add_two_maybe(t_lazy, maybe_false_lazy)
+print(lazy_result)
+assert(lazy_result.cpu(), add_two_maybe(t, maybe_false))
+```
+
+This works as expected! Let's try the case jit trace couldn't handle.
+
+```
+maybe_true_lazy = torch.BoolTensor([1]).to(dev)
+lazy_result = add_two_maybe(t_lazy, maybe_true_lazy)
+assert(lazy_result.cpu(), add_two_maybe(t, maybe_true))
+```
+
+Woohooo! This works too!
+Unfortunately, this flexibility comes with a few downsides. Remember that backends need to translate aten ops into some much lower-level operations that an accelerator understands. The translation process may be time-consuming. Although, usually, it's well worth it! 
+
+TODO: Spiel about awesomeness of whole graph compilations 
```

Comment:
lol.  really? maybe we don't need that here, i think it digresses the main point

Reply:
no spiel 😢